lib/ansible/modules/crypto/luks_device.py

```diff
@@ -744,15 +741,32 @@ def run_module():
     # luks add key
     if conditions.luks_add_key():
         if not module.check_mode:
+            # check if key already exists
+            key_exists = False
+            slotname = str(os.stat(conditions.device).st_mtime)
             try:
-                crypt.run_luks_add_key(conditions.device,
-                                       module.params['keyfile'],
-                                       module.params['passphrase'],
-                                       module.params['new_keyfile'],
-                                       module.params['new_passphrase'])
+                key_exists = crypt.run_luks_open(conditions.device,
+                                                 module.params['new_keyfile'],
+                                                 module.params['passphrase'],
+                                                 slotname)
+            except ValueError:
+                pass
```

Comment:
This is dangerous. Three could be several reasons why opening the container fails. These do not mean that the key does not yet exist.